### PR TITLE
feat: Enhanced LoadCollector with comprehensive testing

### PR DESCRIPTION
**Pull request by jra3**
## Summary
- Enhanced LoadCollector with better error handling, documentation, and graceful degradation
- Added comprehensive test suite with extensive coverage of edge cases
- Improved BaseCollector infrastructure with Logger() method

## Changes
- **BaseCollector**: Added Logger() method for consistent logging patterns
- **LoadCollector**: Enhanced error messages, graceful uptime degradation, comprehensive documentation
- **Tests**: Complete test suite covering constructor, file handling, data parsing, and error scenarios

## Test Coverage
- Constructor tests for different proc path configurations
- Missing file scenarios with graceful degradation validation
- Extensive data parsing tests for valid/invalid inputs
- Organized test constants and helper functions
- Specific error message assertions for better debugging

## Test Plan
- [x] All existing tests continue to pass
- [x] New LoadCollector tests cover edge cases and error scenarios
- [x] Graceful degradation works when uptime is unavailable
- [x] Error messages provide clear context for debugging

### CODE REVIEW

File: pkg/performance/collectors/load_test.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+	config := performance.CollectionConfig{
+		HostProcPath: tmpDir,
+	}
+	collector, err := collectors.NewLoadCollector(logr.Discard(), config)
+	require.NoError(t, err)
+	return collector
+}
+
+func validateCollectorInterface(t *testing.T, collector interface{}) {
```
Comment (haq204) [inline pkg/performance/collectors/load_test.go:58]: Since we added the compile time check we don't need this test

**Review by haq204 — APPROVED** [review]
